src/pybind/rados.py

```diff
@@ -1508,6 +1509,22 @@ def get_locator_key(self):
         return self.locator_key
 
 
+    @requires(('snap', Snap))
+    def set_read(self, snap):
+        """
+        Set the snapshot for reading objects.
+
+        :param snap: the Snap object describing the snapshot to use
+        :type snap: Snap
+
```

Comment:
Should note that you can pass LIBRADOS_SNAP_HEAD to stop reading from a snapshot for snap_id (and define that constant near the top of this file)